pkg/oauth/user.go

```diff
@@ -207,16 +211,49 @@ func (s *server) Login(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
+var errInvalidLogoutRedirectURI = errors.DefinePermissionDenied("invalid_logout_redirect_uri", "the redirect URI did not match the one(s) defined in the client")
+var errMissingAccessTokenIDParam = errors.DefinePermissionDenied("missing_param_access_token_id", "Access token ID was not provided")
+
 func (s *server) Logout(c echo.Context) error {
 	ctx := c.Request().Context()
-	session, err := s.getSession(c)
+	accessTokenID := c.QueryParam("access_token_id")
+	if accessTokenID == "" {
+		return errMissingAccessTokenIDParam
+	}
+	at, err := s.store.GetAccessToken(ctx, accessTokenID)
 	if err != nil {
 		return err
```

Comment:
The access token may already be deleted by the OAuth client at this point, right? Then we would get a NotFound error here, while the end result (logging out) would still be successful.